pkg/pillar/pidfile/pidfile.go

```diff
@@ -27,20 +27,17 @@ func writeMyPid(filename string) error {
 	return ioutil.WriteFile(filename, b, 0644)
 }
 
-func CheckAndCreatePidfile(log *base.LogObject, agentName string) error {
+// CheckNoProcess returns error if agent process is still running
+func CheckNoProcess(log *base.LogObject, agentName string) error {
```

Comment:
I have to stare at this to determine in which case it returns what.

I think that could be avoided if the function is CheckProcessExists(log, agentName) (bool, error)
which returns true if the process exists and false otherwise (and the error is for the caller to log things).

Reply:
Good point, modified